benchmarks/BoltzmannWealth/boltzmann_wealth.py

```diff
@@ -46,13 +46,15 @@ def __init__(self, seed=None, n=100, width=10, height=10):
         self.datacollector = mesa.DataCollector(
             model_reporters={"Gini": compute_gini}, agent_reporters={"Wealth": "wealth"}
         )
-        # Create agents
-        for _ in range(self.num_agents):
-            a = MoneyAgent(self)
-            # Add the agent to a random grid cell
-            x = self.random.randrange(self.grid.width)
-            y = self.random.randrange(self.grid.height)
-            self.grid.place_agent(a, (x, y))
+
+        positions = [
+            (
+                self.random.randrange(self.grid.width),
```

Comment:
What is the conclusion to the discussion of conforming to [SPEC 7](https://scientific-python.org/specs/spec-0007/)? They recommend using NumPy's RNG, and recommend against using seed or random state. Though I don't know how to display/serialize into string a `Generator` object in the SolaraViz.

Reply:
This is an interesting question although not directly relevant for this PR. 

We haven't resolved that conversation but with #1940 merged, we could quite easily add model.np_random or something to that effect in line with SPEC-7. The only drawback is that it would add numpy as a dependency but with pandas also as a dependency that does not really matter. 

Changing `seed` or `random` to rn as keyword throughout the code base is probably something we should do now before 3.0 is stable. @EwoutH, what do you think?

Reply:
It's relevant in the sense that if the RNG is from NumPy, then the positions can be generated in 1 line by specifying the proper shape, and thus making it explicit that having a bulk `create_agents` could simplify the code.

Reply:
Using `rng` instead of `random` sounds as a good idea.

`seed` is a number right, and not a generator?

Reply:
I would be in favor of adding or even switching to numpy.random throughout the entire code base for various reasons, one of which is indeed the easy creation of ndarrays with random numbers. Mesa 3.0 is the best timing to make that change.

Reply:
> Using rng instead of random sounds as a good idea.
> 
> seed is a number right, and not a generator?

In my reading of SPEC-7, `rng` is used as a keyword also for seeding. It can be either a `numpy.random.Generator` or a valid argument for creating such a generator via `numpy.random.default_rng`.